### ISSUE TITLE
[IDE Theme Request] Are there plans for a theme for Eclipse?

### ISSUE THREAD

**Issue by ProxyNeko** [doki-theme/doki-master-theme#17]
I wasn't sure if this would be a suitable place for the request so my apologies if it's not.
Do you have any plans to release this theme set on the Eclipse Marketplace for Eclipse Java IDE users by chance?

Thanks for the hard work you've put into these themes they have been wonderful to use 👍

**Comment by Unthrottled** [comment]
> I wasn't sure if this would be a suitable place for the request 

Sure, it's close enough, I guess. You can always send me an email, I'll probably respond to that as well. However, we can use this issue since it's here.

> Do you have any plans to release this theme set on the Eclipse Marketplace

:thinking: I can say that I've had the idea. I have also been asked before. I know how to add the colors/syntax highlighting to Eclipse. So I know that can be done. However I'll need to figure out how to overlay the stickers on the IDE. I don't know about you, but I **need** the stickers. Sure the colors are nice and match the character, but it's not compete without that last feature. 

I'll take a peek sometime in the near future and see what Eclipse plugins are capable of. If they have access to add/modify things to the UI, then it may be possible to hack something together. 

Thanks for your interest and support!

**Comment by ProxyNeko** [comment]
That's all I can ask, that you'd try is a great thing in itself 😸

I agree with the stickers though, I know a few people who are mega fans of Megumin and some of the other stickers the same as I am but I won't be heartbroken if it can't be done.
 I'll just print it out and stick it on my screen 😆

**Comment by Unthrottled** [comment]
So I did look into what the plugins could do. I think it may be possible to include the sticker after all. It may just take a bit of extra steps to get it set up.

I've got it set up as the next thing that I will work on. I'll make the repo public and close this issue once it is ready. 

Maybe this will make me want to see what Eclipse has to offer that Intellij cannot :thinking:

**Comment by Unthrottled** [comment]
> I'll just print it out and stick it on my screen

Yah know, that probably would have been the easiest thing I could have done. Might have saved me some trouble :joy: 

![Big Brain Fix](https://vignette.wikia.nocookie.net/theyayapocalypse/images/6/65/Galaxybrain.png/revision/latest/window-crop/width/200/x-offset/118/y-offset/0/window-width/567/window-height/566?cb=20190827154717)

**Comment by ProxyNeko** [comment]
That's great! I'll keep my eyes open and look forward to whatever you manage to do but it sounds like you've managed to figure out a lot of it already in some way hahaha

Eclipse was the tool that I used years and years ago but I moved over to IntelliJ for some of the other features, but I'm getting a little fed up of IntelliJ not working with certain checkstyle configs when they work everywhere else, resources for projects not being picked up when using Gradle and other small bugs that wind me up along with the fact that it lags out a fair bit when using a setup similar to Eclipses "Problems" tab. That's the one thing that's drawing me back to Eclipse along with the bugs as of late... But we will see if that improves hopefully 😆

Still thanks for looking into this it gives me more motivation to stick to it while trying it out to see what I like more and what's less buggy for what I do 😸

**Comment by Unthrottled** [comment]
Hey! I figured I give an update. I hit a milestone. So I now have the ability to show a sticker anywhere in the window (it is a view that can be placed anywhere).

![image](https://user-images.githubusercontent.com/15972415/90316855-bba3a900-deea-11ea-989e-7fe0c9bb4901.png)

I plan on having the theme customization set in the preferences.

![image](https://user-images.githubusercontent.com/15972415/90316904-2ead1f80-deeb-11ea-867f-b4dc942afc69.png)

So far so good. I've got the most important feature done. Now I just have to figure out how to bend the Eclipse theme API and Syntax Coloring to my will. So that it doesn't look like garbage.

**Comment by ProxyNeko** [comment]
Lookin good! Apologies for the slow reply, I saw the email while I was half asleep and forgot until now... Glad things are going well though

I did want to ask if you've found a way to change the in use and unused window bar colour up at the top as I've noticed with some themes in eclipse, when you don't have the window in focus it changes from whatever your theme colour is to white and when in focus it's your theme's colour that's the issue I've had with other themes and assumed it might be a limitation of said theme but is that the case?

**Comment by Unthrottled** [comment]
>  in use and unused window bar colour up at the top 

Hmm, are you talking about the tool-bar up at the top with `File Edit Navigate...` ?

I've still got stuff to figure out so what you see is not the final product, but it does not look like it goes to white. (I click outside and stuff)
![GifMaker_20200817184916074](https://user-images.githubusercontent.com/15972415/90455086-a7160b00-e0ba-11ea-86ba-5ebbfbc2da87.gif)

The bar with the window control is always white
![image](https://user-images.githubusercontent.com/15972415/90455116-c2811600-e0ba-11ea-8ef8-406cbdcdfc1d.png)

Bear in mind that this is Windows 10, I really haven't tested anything on Linux or MacOS yet.


Maybe a screen capture would help me understand?

**Comment by ProxyNeko** [comment]
That is the bar I mean right there, I'm using Linux right this second so I couldn't provide a screenshot but that last image is a great example.

![image](https://user-images.githubusercontent.com/19393068/90455500-240ea800-e0ee-11ea-9c96-72c560d1afc0.png)
The bar with minimize and close on with the eclipse logo, I think I used a Dark theme I found on the marketplace when I tested it out but when I went to use another program it turned white but while in use it was black, I'll try to remember to boot into Windows tomorrow and take some screenshots to help a bit more :D

**Comment by Unthrottled** [comment]
No worries!

What theme do you have installed, `DevStyle` ?

I'm also at the point where I am working on trying to get a lot of the small stuff figured out. Like border colors, and getting the syntax themes to work. Which is really hard and time consuming, because I have to look at Eclipse source code/other plugins to try to reverse engineer stuff. So it's going to take me longer to get something that is acceptable (I hope it can be done).

I was playing around with an idea of a quicker/easier short-term fix. Like just producing a DevStyle color import, and just supplying a plugin that supports the stickers. It's more manual, but it could get something in your hands quicker. Just thinking out loud.

**Comment by ProxyNeko** [comment]
Devstyle Dark at the moment yes, I'm in no rush at all take the time you need to get it done to your satisfaction, I know the feeling of wanting to do a good job of something and not rush it out the door haha 

I've also thought of a few theme suggestions if you'd like me to create a new issue for those or post them elsewhere if you have a place for that already :)

**Comment by Unthrottled** [comment]
Cool! I will definitely need the time to figure stuff out. I'll probably heavily inspect Devstyle.

> I've also thought of a few theme suggestions

I have some ideas too! 
You can submit an issue here, I've got a template set up for that now.
I'm curious about other peoples opinions. However you may or may not get what you ask for, but thanks for your interest regardless.

I was playing around with the idea of picking from:

## Ishtar

![Ishtar](https://64.media.tumblr.com/6f26ce4e39dffcdcbb96a8e1ffadbfdd/tumblr_pyx63hkXEv1uzwbyjo1_540.gifv)

## Chika Fujiwara

![Chika](https://64.media.tumblr.com/0c24a45f6f642c16d64ab3540c27b6d4/tumblr_pmqmeqQluI1vmiunio3_250.gifv)

## Aqua

![Aqua](https://i.imgur.com/mSQBDN1.gif)

## Tatsumaki

![Tatsumaki](https://giffiles.alphacoders.com/130/13071.gif)

## Rin Tohsaka

![Rin](https://static.zerochan.net/Tohsaka.Rin.full.2863517.jpg)

**Comment by ProxyNeko** [comment]
I'm all for the following 10000000% 
Rin Tohsaka SO CUTE!, Aqua, Chika, and Ishtar is another OMG SO KAWAI! (If I spelled that right lol) I love the Fate stuff even if it's so complex at times. Never watched One Punch though but I know Tatsumaki is from that at least.

I was gonna suggest some of the girls from New Game the Anime but I feel like they are kinda unimpressive compared to Rin/Ishtar xD 

New Game!
![image](https://user-images.githubusercontent.com/19393068/91625558-4337f000-e9a0-11ea-8de3-eda545f4208d.png)

**Comment by Unthrottled** [comment]
@ProxyNeko 

Hello friend, I did not forget about you!

I did get distracted by some other things though.

I was able to follow through and add: Istar, Rin, Aqua, and Rory :confetti_ball: 

I also have been working on bringing Anime Memes to JetBrains IDEs
https://github.com/waifu-motivator/waifu-motivator-plugin 

(Another reason to choose Intellij over Eclipse :wink:)

Anyways, the repository is now publicly available. 

https://github.com/doki-theme/doki-theme-eclipse

If you have any problems with the plugin, please feel free to raise an issue there.

Thanks for the interest!

**Comment by ProxyNeko** [comment]
Don't worry about it :D I've been a little busy the last few months myself sadly, I did see your new plugin and the extra themes and have to say they look great! I only have one issue with the plugin but I'll raise that on the other repo.

I was a little worried about the Rin theme however, should I open a new issue for that? The errored imports in Java are red but even working code can be red because of the theme xD

**Comment by Unthrottled** [comment]
> I was a little worried about the Rin theme however, should I open a new issue for that? The errored imports in Java are red but even working code can be red because of the theme xD

Lol, yeah I noticed that when I switched to Rin as well. I've fixed it here:

https://github.com/doki-theme/doki-theme-jetbrains/pull/289

The release just went public today
